SphinxDoctest is not sortable

Pytest now runs `tests.sort()` which results in an error complaining that:

```
TypeError: '<' not supported between instances of 'SphinxDoctest' and 'SphinxDoctest'
```

Adding the following comparisons to the `SphinxDoctest` class may resolve this issue:

Operator | Method
-- | --
`==` | `__eq__`
`!=` | `__ne__`
`<` | `__lt__`
`<=` | `__le__`
`>` | `__gt__`
`>=` | `__ge__`

Facing the same issue with cpython 3.7. 

Stacktrace: 
```
../../../.virtualenvs/geopy/lib/python3.7/site-packages/pytest_sphinx.py:368: in collect
    for test in finder.find(module, module.__name__):
../../../.pyenv/versions/3.7.0/lib/python3.7/doctest.py:937: in find
    tests.sort()
E   TypeError: '<' not supported between instances of 'SphinxDoctest' and 'SphinxDoctest'
```
The `tests.sort()` reference in the doctest module: https://github.com/python/cpython/blob/d5fbe9b1a3d65ceeb9159c5ba999ee966a945f76/Lib/doctest.py#L937

The comparison operator defined in `DocTest` (which instances are being sorted): https://github.com/python/cpython/blob/d5fbe9b1a3d65ceeb9159c5ba999ee966a945f76/Lib/doctest.py#L561-L566

BTW, is there a reason to introduce `SphinxDoctest` instead of just using the `doctest.DocTest`?

--- a/pytest_sphinx.py
+++ b/pytest_sphinx.py
@@ -37,18 +37,6 @@ class SphinxDoctestDirectives(enum.Enum):
     DOCTEST = 5
 
 
-class SphinxDoctest(object):
-    def __init__(self, examples, docstring,
-                 filename='<sphinx-doctest>'):
-        super(SphinxDoctest, self).__init__()
-        self.examples = examples
-        self.globs = {}
-        self.name = None
-        self.lineno = None
-        self.filename = filename
-        self.docstring = docstring
-
-
 def pytest_collect_file(path, parent):
     config = parent.config
     if path.ext == ".py":
@@ -109,10 +97,10 @@ def _find_options(want, name, lineno):
     return options
 
 
-def docstring2test(docstring):
+def docstring2examples(docstring):
     """
     Parse all sphinx test directives in the docstring and create a
-    SphinxDoctest object.
+    list of examples.
     """
     # TODO subclass doctest.DocTestParser instead?
 
@@ -121,7 +109,7 @@ def docstring2test(docstring):
                any(line.startswith('.. ' + d.name.lower() + '::')
                    for d in SphinxDoctestDirectives)]
     if not matches:
-        return SphinxDoctest([], docstring)
+        return []
 
     matches.append(len(lines))
 
@@ -184,7 +172,7 @@ def docstring2test(docstring):
                                 lineno=y.lineno - 1,
                                 options=options))
 
-    return SphinxDoctest(examples, docstring)
+    return examples
 
 
 class SphinxDocTestRunner(doctest.DebugRunner):
@@ -324,11 +312,12 @@ class SphinxDocTestRunner(doctest.DebugRunner):
 class SphinxDocTestParser(object):
     def get_doctest(self, docstring, globs, name, filename, lineno):
         # TODO document why we need to overwrite? get_doctest
-        test = docstring2test(docstring)
-        test.name = name
-        test.lineno = lineno
-        test.filename = filename
-        return test
+        return doctest.DocTest(examples=docstring2examples(docstring),
+                               globs=globs,
+                               name=name,
+                               filename=filename,
+                               lineno=lineno,
+                               docstring=docstring)
 
 
 class SphinxDoctestTextfile(pytest.Module):
@@ -346,9 +335,12 @@ class SphinxDoctestTextfile(pytest.Module):
                                      optionflags=optionflags,
                                      checker=_pytest.doctest._get_checker())
 
-        test = docstring2test(text)
-        test.name = name
-        test.lineno = 0
+        test = doctest.DocTest(examples=docstring2examples(text),
+                               globs={},
+                               name=name,
+                               filename=name,
+                               lineno=0,
+                               docstring=text)
 
         if test.examples:
             yield _pytest.doctest.DoctestItem(

--- a/tests/test_doc2test.py
+++ b/tests/test_doc2test.py
@@ -1,6 +1,6 @@
 import doctest
 
-from pytest_sphinx import docstring2test
+from pytest_sphinx import docstring2examples
 
 
 def test_simple():
@@ -16,9 +16,9 @@ def test_simple():
      '5': 6}
 """
 
-    test = docstring2test(doc)
-    assert len(test.examples) == 1
-    example = test.examples[0]
+    examples = docstring2examples(doc)
+    assert len(examples) == 1
+    example = examples[0]
 
     assert example.want == "{'3': 4,\n '5': 6}\n"
     assert example.exc_msg is None
@@ -38,12 +38,12 @@ def test_with_options():
     {'3': 4,
      '5': 6}"""
 
-    test = docstring2test(doc)
-    assert len(test.examples) == 1
-    example = test.examples[0]
+    examples = docstring2examples(doc)
+    assert len(examples) == 1
+    example = examples[0]
 
     assert example.want == "{'3': 4,\n '5': 6}\n"
-    assert (docstring2test(doc + '\n').examples[0].want
+    assert (docstring2examples(doc + '\n')[0].want
             == "{'3': 4,\n '5': 6}\n")
     assert example.exc_msg is None
     assert example.options == {

--- a/tests/test_python_files.py
+++ b/tests/test_python_files.py
@@ -125,3 +125,36 @@ def test_working_module_doctest_nospaces(testdir):
     result = testdir.runpytest('--doctest-modules')
     result.stdout.fnmatch_lines([
         '*=== 1 passed in *'])
+
+
+def test_multiple_doctests_in_single_file(testdir):
+
+    testdir.makepyfile(textwrap.dedent("""
+        def foo():
+            \"\"\"
+            .. testcode::
+
+                print('1adlfadsf')
+
+            .. testoutput::
+
+                1...
+            \"\"\"
+            pass
+
+        def bar():
+            \"\"\"
+            .. testcode::
+
+                print('1adlfadsf')
+
+            .. testoutput::
+
+                1...
+            \"\"\"
+            pass
+    """))
+
+    result = testdir.runpytest('--doctest-modules')
+    result.stdout.fnmatch_lines([
+        '*=== 2 passed in *'])

